datapipes/aws_lambda/cloudwatch_to_firehose/naboo_preprocessor.py

```diff
@@ -0,0 +1,204 @@
+
+    return json_log
+
+
+def log_has_tags_of_interest(tags, line):
+    """
+    Check if this line has one of registered tags. This search is exclusive.
+
+    :param tags: Array of strings that we want to look for in the log line
+    :param line: A log line
+    :return: The tag first found in the log line. None, otherwise.
+    """
+    for tag in tags:
+        if tag in line:
+            return tag
+
+    return None
+
+
+def lambda_handler(event, context):
+    """
+    :param event:
+        In the format of:
+        {
+            "awslogs": {
+                "data": "H4sIAAAAAAAAAO1c227jOBL9FcPPiYdkkSwywDw0....[gzipped base64 string]"
+        }
+    :param context:
+    :return: None
+    """
+
+    if not event['awslogs']['data']:
+        return
+
+    stream = event['awslogs']['data']
+    # TODO: data stream is not being chunked under python 3.6
+    payload = json.loads(zlib.decompress(base64.b64decode(stream), 32 + zlib.MAX_WBITS).decode('utf-8'))
+    records = dict((t, []) for t in log_config.keys())
+
+    # TODO: change "test-stream" to production stream name before it's deployed
+    if not payload["logStream"] or "test-stream" != payload["logStream"]:
+        return
+
+    extract_and_push_records(records, payload)
+    flush_records(records)
+    return True
+
+
+def extract_and_push_records(records, payload):
+    """
+    loop through every log event within the event log, extract the actual log message, reformat into compatible
+    format and push to firehose
+
+    :param records:
+        In the format of:
+        {
+            'response_log=': [],
+            'feed_event=': [
+                {
+                    'Data': '{"ts":"2018-03-13T01:12:17.605578Z","props":{"watched_till":21,"video_id":79,"user_id":31,
+                        "inserted_at":"2018-03-13T01:12:17.605538Z","id":null,"duration":23,"completed":false},
+                        "event":"engage_video"}'
+                }
+            ]
+        }
+    :param payload:
+        In the format of:
+        {
+            'messageType': 'DATA_MESSAGE',
+            'owner': '170553093583',
+            'logGroup': 'shawn-test',
+            'logStream': 'test-stream',
+            'subscriptionFilters': ['LambdaStream_naboo-preprocessor'],
+            'logEvents':
+                [
+                    {
+                        'id': '33932602058168217326485955733116808031206164863324984556',
+                        'timestamp': 1521590500949,
+                        'message': '18:12:17.594 [info] feed_event={"ts":"2018-03-13T01:12:17.594577Z","props":
+                            {"video_id":79,"user_id":1,"id":1239},"event":"show_video"}'
+                    },
+                    ...
+                ]
+        }
+    :return:
+    """
+    for log_event in payload['logEvents']:
+        json_log = extract_controller_json_str(log_event['message'])
+
+        if all(json_log):
+            logger.info("====" + str(json_log))
```

Comment:
Time to move to .debug. How about pass in log level as an environment variable like a debug mode?